examples/templates/cluster/small-aws-provider.yaml

```diff
@@ -0,0 +1,49 @@
+---
+APIVersion: v1
+Kind: Provider
+Metadata:
+  name: aws_test
+CloudProvider: aws
+Parameters:
+  locations:
+  - us:
+    - us-west2
+  - others:
+    - southamerica-east1
+    - northamerica-northeast1
+    - australia-southeast1
+  NodeType:
+    x-smal: "t2.nano"
+    small: "t2.micro"
+    mediem: "t2.medium"
+    large: "t2.large"
+    x-large: "t2.xlarge"
+    custom:
+      cpu: ""
+      memory: ""
```

Comment:
I love this. Maybe we could allow people to create their own node definitions:

```
  NodeTypeDefinitions:
    - name: x-smal
       providertype: "t2.nano"
    - name: small
      providertype: "t2.micro"
    - name: elasticsearchhost
      customtype:
        cpu: "100"
        memory: "100"
```

Reply:
That a very good idea